lightning/backend/local.go

```diff
@@ -1156,11 +1247,30 @@ func (local *local) ImportEngine(ctx context.Context, engineUUID uuid.UUID) erro
 	return nil
 }
 
+func (local *local) ResetEngine(ctx context.Context, engineUUID uuid.UUID) error {
+	// the only way to reset the engine + reclaim the space is to delete and reopen it 🤷
+	if err := local.CleanupEngine(ctx, engineUUID); err != nil {
+		return err
+	}
+	if err := local.OpenEngine(ctx, engineUUID); err != nil {
```

Comment:
I think here, it's better to only clean up the engine, and left the lazy open engine operation to write rows. In this way we can avoid the corner case that the extra open engine may never be visited again.
Example(if check disk quota and the normal engine write finish at the same time, and this engine is big enough to trigger the unsafe import):
```
disk-quote-routine                         normal-engine-routine
1. lock-and-import
                                            2. lock-and-import
                                            3. clean-up-engine
4. cleanup-and-reopen
```
In this timeline, after action 4, the reopened engine is never visited again, and it's data files won't be clean up either.

Reply:
changed to do nothing at step 4 if the engine is already gone.